api/src/main/java/org/apache/iceberg/expressions/ManifestEvaluator.java

```diff
@@ -148,10 +148,6 @@ public <T> Boolean isNaN(BoundReference<T> ref) {
         return ROWS_CANNOT_MATCH;
       }
 
-      if (allValuesAreNull(stats.get(pos), ref.type().typeId())) {
```

Comment:
I don't think this is correct. The `containsNaN` field was introduced later, so it can be null for metadata files that were written before we tracked NaN values in partitions. In that case, if we know that all values are null then none of them are NaN and can then eliminate scanning the manifest.

Reply:
Yes, you are right on that one! It didn't think of encoding NaN as None. Then this makes sense